vignettes/col_formatting.Rmd

```diff
@@ -0,0 +1,204 @@
+  disp_cols = c("BMRKR1", "ASEQ", "AESEV"),
+)
+
+lsting_1
+```
+
+Notice that all of the data in the table above is displayed as is, with no rounding or formatting applied. All columns are centered and all missing values are displayed as `"NA"`.
+
+Suppose we want to left align all of the columns in the listing and replace missing values with the string `"<No data>"`. This can be done by setting the `all` element in the list supplied to `default_formatting`, as shown in the following example.
+
+```{r}
+default_fmt <- list(
+  all = fmt_config(na_str = "<No data>", align = "left")
+)
+
+lsting_2 <- as_listing(
+  df = adae,
+  key_cols = c("USUBJID", "AGE", "TRTSDTM"),
+  disp_cols = c("BMRKR1", "ASEQ", "AESEV"),
+  default_formatting = default_fmt
+)
+
+lsting_2
+```
+
+Now consider that we would like to display our numeric columns with two decimal places and then align these columns on the decimal point. This can be done by adding a `"numeric"` element to the `default_formatting` list as follows:
+
+```{r}
+default_fmt <- list(
+  all = fmt_config(na_str = "<No data>", align = "left"),
+  numeric = fmt_config(format = "xx.xx", na_str = "<No data>", align = "decimal")
+)
+
+lsting_3 <- as_listing(
+  df = adae,
+  key_cols = c("USUBJID", "AGE", "TRTSDTM"),
+  disp_cols = c("BMRKR1", "ASEQ", "AESEV"),
+  default_formatting = default_fmt
+)
+
+lsting_3
+```
+
+Along with the format strings listed by `formatters::list_valid_format_labels`, we can also specify a format _function_ to allow for more customized formats in our listing. In the following example, we will define and apply a custom format function to format date (`POSIXt` class) columns in our listing.
+
+```{r}
+# Custom format function - takes date format as input
+date_fmt <- function(fmt) {
+  function(x, ...) do.call(format, list(x = x, fmt))
+}
+
+default_fmt <- list(
+  all = fmt_config(na_str = "<No data>", align = "left"),
+  numeric = fmt_config(format = "xx.xx", na_str = "<No data>", align = "decimal"),
+  POSIXt = fmt_config(format = date_fmt("%B %d, %Y @ %I:%M %p %Z"), na_str = "<No data>")
+)
+
+lsting_4 <- as_listing(
+  df = adae,
+  key_cols = c("USUBJID", "AGE", "TRTSDTM"),
+  disp_cols = c("BMRKR1", "ASEQ", "AESEV"),
+  default_formatting = default_fmt
+)
+
+lsting_4
+```
+
+In the output above, the `all` format configuration, which originally applied to all columns in the listing, now only applies to the two character/factor variables (`USUBJID` and `AESEV`). This is because all other data classes in the listing have been covered by other elements in the list provided to `default_formatting`. When format configurations are supplied to a listing, any other applicable configuration take precedence over the `all` format configuration.
+
+## Column-Wise Formatting in `as_listing`
+
+In this section we will demonstrate how custom formatting can be applied on a column-by-column basis rather than to all columns of a specified data class or an entire listing at once. 
```

Comment:
```suggestion
In this section, we will demonstrate how custom formatting can be applied on a column-by-column basis rather than to all columns of a specified data class or an entire listing at once. 
```